app/services/permissions_checker.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class PermissionsChecker
+  def initialize(current_user, permission_name)
+    @current_user = current_user
+    @permission = Permission.find_by(name: permission_name)
+  end
+
+  def call
+    return true if @current_user.role.role_permissions.find_by(permission_id: @permission.id).value == 'true'
```

Comment:
Just learned that this is actually firing multiple queries.

I believe that something like this is the preferred method (one query):
 ```
RolePermission.joins(:permission)
  .where(provider: 'greenlight')
  .where(role_id: current_user.role_id)
  .find_by(permission: { name: @permission.name }).value == 'true' 
```

Reply:
Indeed, this query is almost twice as fast